osf/models/contributor.py

```diff
@@ -80,6 +80,22 @@ class Meta:
         unique_together = ('user', 'institution')
 
 
+class DraftRegistrationContributor(AbstractBaseContributor):
+    draft_registration = models.ForeignKey('DraftRegistration', on_delete=models.CASCADE)
```

Comment:
ideally, this foreign key field to draft_registration would be named `resource`, and likewise, corresponding fields on other contributor tables (node/preprint) would also be named `resource`.  This would make the ContributorMixin code a lot cleaner.